app/scripts/controller-init/confirmations/transaction-controller-init.ts

```diff
@@ -247,6 +260,62 @@ function publishSmartTransactionHook(
   });
 }
 
+function publishBatchSmartTransactionHook({
+  transactionController,
+  smartTransactionsController,
+  hookControllerMessenger,
+  flatState,
+  transactions,
+}: {
+  transactionController: TransactionController;
+  smartTransactionsController: SmartTransactionsController;
+  hookControllerMessenger: SmartTransactionHookMessenger;
+  flatState: ControllerFlatState;
+  transactions: PublishBatchHookTransaction[];
+}) {
+  // UI state is required to support shared selectors to avoid duplicate logic in frontend and backend.
+  // Ideally all backend logic would instead rely on messenger event / state subscriptions.
+  const uiState = getUIState(flatState);
```

Comment:
This is very similar to the other hook, so pending a future refactor to just use the messenger, we could have a general function here to remove some duplication?